cmd/provider/main.go

```diff
@@ -37,11 +38,10 @@ func main() {
 	var (
 		app              = kingpin.New(filepath.Base(os.Args[0]), "Terraform based Crossplane provider for Template").DefaultEnvars()
 		debug            = app.Flag("debug", "Run with debug logging.").Short('d').Bool()
-		syncPeriod       = app.Flag("sync", "Controller manager sync period such as 300ms, 1.5h, or 2h45m").Short('s').Default("1h").Duration()
+		syncInterval     = app.Flag("sync", "Sync interval controls how often all resources will be double checked for drift.").Short('s').Default("1h").Duration()
```

Comment:
Hi @haarchri,
The sync interval has a rather specific meaning tied to the internals of the controller-runtime:
https://github.com/kubernetes-sigs/controller-runtime/blob/304027bcbe4b3f6d582180aec5759eb4db3f17fd/pkg/cache/cache.go#L146

So, I would personally prefer keeping the reference to the controller-runtime in the explanation. There's also a warning there:
```
Change this value only if you know what you are doing. Defaults to 10 hours if unset.
```

I remember us discussing the name of this parameter and I remember saying `syncInterval` is a better alternative than the `syncPeriod`. Looks like we have borrowed the name from the upstream configuration parameter we are overriding. I've also seen that the core Crossplane documents it in the way suggested in this PR. 

Even though we do the renaming to be consistent in the ecosystem, I believe we had better continue mentioning the controller-runtime and warn users that this is rather an advanced parameter to tackle with:
```suggestion
		syncInterval     = app.Flag("sync", "Controller manager cache sync period such as 300ms, 1.5h, or 2h45m. Determines the minimum frequency at which watched resources are reconciled. Use with caution.").Short('s').Default("1h").Duration()
```